cli/pull.ml

```diff
@@ -65,9 +70,25 @@ let run (`Yes yes) (`Root root) (`Lockfile explicit_lockfile)
         Common.filter_duniverse ~to_consider:duniverse_repos duniverse
       in
       let* () = check_dune_lang_version ~yes ~root in
-      OpamGlobalState.with_ `Lock_none (fun global_state ->
-          Pull.duniverse ~global_state ~root ~full
-            ~trim_clone:(not keep_git_dir) duniverse)
+      OpamGlobalState.with_ `Lock_none @@ fun global_state ->
+      let* locked_ocaml_version =
+        Lockfile.ocaml_version lockfile
+        |> Result.of_option ~error:(`Msg "OCaml compiler not in lockfile")
+      in
+      OpamSwitchState.with_ `Lock_none global_state @@ fun switch_state ->
+      let switch_ocaml_version =
+        OpamSwitchState.get_package switch_state Config.compiler_package_name
+        |> OpamPackage.version
+      in
+      (match version_is_at_least locked_ocaml_version switch_ocaml_version with
+      | true -> ()
+      | false ->
+          Logs.warn (fun l ->
```

Comment:
How about printing the warning after the pull? I think it will make it clearer that the pull was successful and that this requires folloe up actions by the user rather than it being an opam-monorepo failure.

Reply:
I think this is a good idea indeed! Also makes the warning more visible.